### ISSUE TITLE
SRWLock implementation still disabled ?

### ISSUE THREAD

**Issue by Carandiru0** [uxlfoundation/oneTBB#185]
Looks like siince 2012:
https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/327962

SRWLocks are not being used on Windows.
In windows_api.h at the end of the file:

#if defined(RTL_SRWLOCK_INIT)
#ifndef __TBB_USE_SRWLOCK
// TODO: turn it on when bug 1952 will be fixed
#define __TBB_USE_SRWLOCK 0
#endif
#endif

Looks like this one was missed!

I'm guessing that this was not intentional for this long period of time.

**Comment by anton-potapov** [comment]
It seems that oneTBB still do not use SRWLOCK by default

**Comment by arunparkugan** [comment]
@anton-potapov is this issue still relevant?

**Comment by anton-potapov** [comment]
@arunparkugan , I believe it is a question to issue creator - @Carandiru0 :)

**Comment by e4lam** [comment]
Yes, this is still relevant because SRW locks are best for performance on Windows. In the CRT, std::mutex itself uses SRW locks underneath.

**Comment by Carandiru0** [comment]
Not considered. Should effectively be portable.

**Comment by e4lam** [comment]
@Carandiru0 Not following. The gist of this issue is that we're leaving performance on the floor that's easy to have.
